openmdao/core/group.py

```diff
@@ -613,6 +613,28 @@ def apply_nonlinear(self, params, unknowns, resids, metadata=None):
                 else:
                     sub.apply_nonlinear(sub.params, sub.unknowns, sub.resids, metadata)
 
+    def sys_jacobian(self, params, unknowns, resids, total_derivs=True): 
```

Comment:
it is needed currently because it it changes the default value of the total_derivs argument from False (in system) to True in (Group). 

However, this is a bit of a hack. It prevented me from having to test for group-vs-component in the base class, but its not super clean. I'll refactor a bit more to clean this up